ekncontent/overrides/EosKnowledgeContent.js

```diff
@@ -89,7 +89,9 @@ function _init() {
     });
     define_property(Eknc.ArticleObjectModel, 'table-of-contents', {
         get: function () {
-            let toc = this.get_table_of_contents().deep_unpack();
+            let toc = this.get_table_of_contents();
+            if (toc)
+                toc = toc.deep_unpack();
```

Comment:
Well the gvariant there is gauranteed to be a array. So if its not null, the case you are talking about is where its any empty array right? If that case the forEach below would be a no-op so an early return is fine by not necessary.

Even if we do need extra check for a reason I am missing I'd write everything with early retruns
```
if (!toc)
  return [];
toc = toc.deep_unpack();
if (!toc)
  return [];
```

But do think we can just ditch the second if.